### PR TITLE
logger docs and api docs

### LINKED ISSUE #3759: Tensorboard, what does epoch chart show?
Fast question. 

Image from Tensorboard:
image

I don't add this log. It's done automatically by lightning. What does this chart present?

### PR DESCRIPTION
**Pull request by ydcjeff**
## What does this PR do?

Follow up PR of #3908 

Fixes #3759 

The previous PR uses `autosummary` for callbacks
After using callbacks, some links, like :class:`~pytorch_lightning.trainer.trainer.Trainer` and paramref no longer work.
To fix that, we will generate a API References docs to link and users who like to read only API docs can read too.

So, what's different with old past apidocs?

- the current API references page uses `autosummary` with the `module.rst` template in `autosummary` folder.
- we can only include the APIs (modules) that we want users to use.

# Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [x] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? 
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/core/hooks.py [hunk 1]
```diff
@@ -12,6 +12,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+"""Various Hooks to be used in the Lightning code."""
```
Comment (awaelchli) [inline pytorch_lightning/core/hooks.py:15]:
```suggestion
"""Various hooks to be used in the Lightning code."""
```

**Review by awaelchli — COMMENTED** [review]
image


@ydcjeff the logger summary table includes parts of the full docstring and it cuts off. Do you know how to just show the short title?

**Comment by ydcjeff** [comment]
> image
> 
> @ydcjeff the logger summary table includes parts of the full docstring and it cuts off. Do you know how to just show the short title?

autosummary parses the first sentence of docstring, so updated the docstring.

File: docs/source/api_references.rst [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+.. autosummary::
+    :toctree: api
+    :nosignatures:
+
+    datamodule
+    decorators
+    hooks
+    lightning
+    step_result
```
Comment (awaelchli) [inline docs/source/api_references.rst:17]: Should we remove step result? Since it is now deprecated?

**Review by awaelchli — APPROVED** [review]

**Comment by ydcjeff** [comment]
Btw, how do you feel like new API Ref page?
Needs to finalize ofc

**Review by justusschock — APPROVED** [review]

**Comment by awaelchli** [comment]
@edenlightning this PR adds a new section "API references" in the navigation panel. 
See here
https://66112-178626720-gh.circle-artifacts.com/0/html/api_references.html
I faintly remember you asking for this a while ago, or was it something else?

**Review by SkafteNicki — APPROVED** [review]
LGTM

**Review by Borda — APPROVED** [review]
lgtm

**Review by rohitgr7 — APPROVED** [review]

**Pull request merged**